src/PublicKey/Rsa.php

```diff
@@ -259,7 +259,11 @@ public function encrypt($data, Rsa\AbstractKey $key = null)
             throw new Exception\InvalidArgumentException('No key specified for the decryption');
         }
 
-        $encrypted = $key->encrypt($data);
+        if (null === $this->getOptions()->getPadding()) {
```

Comment:
Don't repeat the call chain to `$this->getOptions()->getPadding()`, as it is not guaranteed to return the same values over subsequent calls anyway. Assign it to a variable instead.